logger.js

```diff
@@ -0,0 +1,116 @@
+function log(message) {
+    console.log(message);
+}
+
+function log2(...msgArgs) {
+    let message = msgArgs.join(" | ");
+    log(message);
+}
+
+function log3(...msgArgs) {
+    let date = new Date();
+    let dateStr = `${date.toLocaleDateString()}, ${date.toLocaleTimeString()}`;
+    log2(dateStr, ...msgArgs);
+}
+
+log("Hello World !");
+log2("Hello", "World", "!");
+log3("Hello", "World", "!");
+
+
+//Functional pattern
```

Comment:
Code below is not the Functional pattern. This code doesn't implement inheritance at all.